src/refnux.coffee

```diff
@@ -50,6 +50,15 @@ createStore = (state = {}) ->
         # execute the action
         try
             newval = action state, dispatch
+
+            if newval and typeof(newval.then) is 'function'
+                dispatching = false
+                return newval.then(
```

Comment:
the previous dispatch function didn't have a defined return. now you're saying it's returning the promise. i don't mind that, but i think we should ensure that also the non-promise variant returns the val. i.e. `return newval` at the end of the function.